modules/metastorage/src/test/java/org/apache/ignite/internal/metastorage/server/BasicOperationsKeyValueStorageTest.java

```diff
@@ -2287,6 +2295,49 @@ void testInvokeAndOperationTimestamp() {
         checkEntriesTimestamp(keys, storage.revision(), timestamp2, timestamp0, timestamp1);
     }
 
+    @Test
+    void testSnapshot() throws Exception {
+        byte[] key = key(0);
+        byte[] value = keyValue(0, 0);
+
+        storage.put(key, value, hybridTimestamp(10));
+
+        Path snapshotDir = workDir.resolve("snapshotDir");
+
+        restartStorage();
+
+        assertEquals(0L, storage.revision());
+        assertEquals(0L, storage.updateCounter());
+        assertTrue(storage.get(key).empty());
+
+        storage.restoreSnapshot(snapshotDir);
+
+        assertEquals(1L, storage.revision());
+        assertEquals(1L, storage.updateCounter());
+        assertFalse(storage.get(key).empty());
+    }
+
+    @Test
+    void testCleatDataBeforeRestoreFromSnapshot() {
+        byte[] key0 = key(0);
+        byte[] key1 = key(1);
+        byte[] value = keyValue(0, 0);
+
+        storage.put(key0, value, hybridTimestamp(10));
+
+        Path snapshotDir = workDir.resolve("snapshotDir");
+        assertThat(storage.snapshot(snapshotDir), willCompleteSuccessfully());
+
+        storage.put(key1, value, hybridTimestamp(10));
+
+        storage.restoreSnapshot(snapshotDir);
+
+        assertEquals(1L, storage.revision());
+        assertEquals(1L, storage.updateCounter());
+        assertFalse(storage.get(key0).empty());
```

Comment:
Why is `key0` empty? Isn't it supposed to be a part of the snapshot?

Reply:
`assertFalse(storage.get(key0).empty());` 
Checks that it is NOT empty, you probably didn't notice. =)

Reply:
Yeah, I missed that, sorry